javav2/example_code/s3/src/main/java/com/example/s3/KMSEncryptionExample.java

```diff
@@ -0,0 +1,240 @@
+   the License is located at
+    http://aws.amazon.com/apache2.0/
+   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+   specific language governing permissions and limitations under the License.
+*/
+
+package com.example.s3;
+
+// snippet-start:[s3.java2.kms.import]
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.kms.KmsClient;
+import software.amazon.awssdk.services.kms.model.DecryptRequest;
+import software.amazon.awssdk.services.kms.model.EncryptRequest;
+import software.amazon.awssdk.services.kms.model.EncryptResponse;
+import software.amazon.awssdk.services.kms.model.KmsException;
+import software.amazon.awssdk.services.kms.model.DecryptResponse;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import java.io.FileNotFoundException;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+import java.io.IOException;
+import java.io.FileInputStream;
+// snippet-end:[s3.java2.kms.import]
+
+/**
+ * Before running this code example, you need to create a key by using the AWS Key Management Service. 
+ * For information, see "Creating keys" in the AWS Key Management Service Developer Guide.
+ */
+public class KMSEncryptionExample {
+
+    public static void main(String[] args) {
+        final String USAGE = "\n" +
+                "Usage:\n" +
+                "    KMSEncryptionExample <objectname><bucketName> <objectPath><outPath><keyId>\n\n" +
+                "Where:\n" +
+                "    objectname - the name of the object \n\n" +
+                "    bucketName - the bucket name that contains the object (i.e., bucket1)\n" +
+                "    objectPath - the path to a TXT file to encrypt and place into a S3 bucket (i.e., C:\\AWS\\test.txt)\n" +
+                "    outPath - the path where a text file is written to after it's decrypted (i.e., C:\\AWS\\testPlain.txt)\n" +
+                "    keyId - the id of the KMS key to use to encrpt/decrypt the data. You can obtain the key ID value from the AWS KMS console\n";
+
+         if (args.length < 5) {
+             System.out.println(USAGE);
+             System.exit(1);
+        }
+
+        /* Read the name from command args*/
+        String objectKey = args[0];
+        String bucketName = args[1];
+        String objectPath = args[2];
+        String outPath = args[3];
+        String keyId = args[4];
+
+        //Create the S3Client object
```

Comment:
All comments should be grammatically complete, including punctuation (here and throughout).

Reply:
fixed